### DISCUSSION TITLE
How do I know which page am I in Layout.js?

### DISCUSSION THREAD

**Discussion by naseef0** [vercel/next.js#51328] [App Router]
Example:
    --layout.js
    --page.js
    ----dashboard.js
    ------page.js
   
in Layout.js: I need to know which page was rendered. 

```
export default function Layout({ children }) {

  const pageType ="Home"

  if(pageType==="Home"){
    <------ do something ----->
  }
  
  if(pageType==="Dashboard"){
    <------ do something ----->
  }

  return (
    <>
      <Navbar />
      <main>{children}</main>
      <Footer />
    </>
  )
}
```

**Comment by market-south** [comment]
Hi, 

Assuming you are using V13

```
'use client'
import './css/globals.css'
import { usePathname } from "next/navigation";
import { useRef } from "react"

export default function RootLayout({ children }) {

 const pageType = usePathname();

  if(pageType === "/" || pageType === "/home"){
    <------ do something ----->
  }
  
  if(pageType ==="/dashboard" ){
    <------ do something ----->
  }

  return (
    <>
      <Navbar />
      <main>{children}</main>
      <Footer />
    </>
  )
}
```

Hope this helps

**Reply by naseef0** [reply]
@market-south using `"use client"` and `usePathname` in layout.js?  is that the correct way?

**Reply by market-south** [reply]
@naseef0  yes in V13 we have to declare the `use client` directive to use the useRef. But if you just need the usePathname you don't have to

**Reply by naseef0** [reply]
@market-south  I just need to know which page was rendered. if we use "use client" in layout.js, Then the entire app will be in client, right? then what is the use of RSC? correct me if I am wrong.

**Reply by market-south** [reply]
@naseef0  The answer is no. Declaring `use client` doesn't set the entire app as a client but the child component(s).

 In your case, you don't need the client directive as you are only using `usePathname` to know which page is rendered. 

If it helps https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive
